dhall/src/Dhall/Map.hs

```diff
@@ -456,27 +454,12 @@ traverseWithKey f m =
     f' (k, a) = fmap ((,) k) (f k a)
 {-# INLINABLE traverseWithKey #-}
 
-{-| Traverse all of the key-value pairs in a `Map`, in their original order
-    where the result of the computation can be forgotten.
-
->>> traverseWithKey_ (\k v -> print (k, v)) (fromList [("B",1),("A",2)])
-("B",1)
-("A",2)
--}
-traverseWithKey_
-    :: Ord k => Applicative f => (k -> a -> f ()) -> Map k a -> f ()
-traverseWithKey_ f m = Data.Functor.void (traverseWithKey f m)
-{-# INLINABLE traverseWithKey_ #-}
-
-{-| Travese all of the key-value pairs in a 'Map', not preserving their
+{-| Traverse all of the key-value pairs in a 'Map', not preserving their
     original order, where the result of the computation can be forgotten.
-
-    Note that this is an optimisation over 'traverseWithKey_' since we do
-    not care in what order we traverse the pairs.
 -}
 unorderedTraverseWithKey_
     :: Ord k => Applicative f => (k -> a -> f ()) -> Map k a -> f ()
-unorderedTraverseWithKey_ f = Data.Functor.void . traverse_ (uncurry f) . toList
+unorderedTraverseWithKey_ f = traverse_ (uncurry f) . toList
```

Comment:
Interestingly, the usage of `toList` means that this is in fact a traversal in the original key order!

Once the benchmarks work again I'll try optimizing this.